packages/esm-patient-registration-app/src/root.component.tsx

```diff
@@ -1,33 +1,41 @@
 import React from 'react';
 import { BrowserRouter, Route } from 'react-router-dom';
 import { Grid, Row } from 'carbon-components-react';
-import { ExtensionSlot } from '@openmrs/esm-framework';
-import { Resources, ResourcesContext } from './offline.resources';
+import { ExtensionSlot, Session } from '@openmrs/esm-framework';
+import {
+  fetchAddressTemplate,
+  fetchAllRelationshipTypes,
+  fetchCurrentSession,
+  fetchPatientIdentifierTypesWithSources,
+  Resources,
+  ResourcesContext,
+} from './offline.resources';
 import { SavePatientForm } from './patient-registration/form-manager';
 import { PatientRegistration, PatientRegistrationProps } from './patient-registration/patient-registration.component';
+import useSWR from 'swr';
 
 export interface RootProps extends PatientRegistrationProps, Resources {
   savePatientForm: SavePatientForm;
   isOffline: boolean;
 }
 
-export default function Root({
-  currentSession,
-  addressTemplate,
-  relationshipTypes,
-  identifierTypes,
-  savePatientForm,
-  isOffline,
-}: RootProps) {
-  const resources = {
-    currentSession,
-    addressTemplate,
-    relationshipTypes,
-    identifierTypes,
-  };
+export default function Root({ savePatientForm, isOffline }: RootProps) {
+  const { data: currentSession } = useSWR<Session, Error>('patientRegistrationCurrentSessionInfo', fetchCurrentSession);
+  const { data: addressTemplate } = useSWR('patientRegistrationAddressTemplate', fetchAddressTemplate);
+  const { data: relationshipTypes } = useSWR('patientRegistrationRelationshipTypes', fetchAllRelationshipTypes);
+  const { data: identifierTypes } = useSWR(
+    'patientRegistrationPatientIdentifiers',
+    fetchPatientIdentifierTypesWithSources,
+  );
 
   return (
-    <ResourcesContext.Provider value={resources}>
+    <ResourcesContext.Provider
```

Comment:
From my side, this doesn't need to happen in this PR, I just want to mention the following:
Since we do migrate to SWR, a good follow-up refactoring/issue to this PR would be to remove this `ResourcesContext` in favor of direct SWR calls in the respective components which require the data that's currently passed down via the context. That would free this `Root` component from doing any data fetching.

But again, from my side, this absolutely doesn't have to happen in this PR as the focus here really can remain on removing the `resources` usage and thus improving UX perf.

Reply:
Sure!